warehouse/query-core/src/main/java/datawave/query/jexl/visitors/ValidQueryPropertyMarkerVisitor.java

```diff
@@ -0,0 +1,38 @@
+package datawave.query.jexl.visitors;
+
+import datawave.query.exceptions.DatawaveFatalQueryException;
+import datawave.query.jexl.nodes.QueryPropertyMarker;
+import org.apache.commons.jexl2.parser.ASTAndNode;
+import org.apache.commons.jexl2.parser.JexlNode;
+
+/**
+ * This visitor validates that all query property marker nodes present have a singular root source node.
+ */
+public class ValidQueryPropertyMarkerVisitor extends BaseVisitor {
```

Comment:
I'm going to make a few requests for this one.
- create a package under datawave/query/jexl/visitors called 'validate'
- move this class to the validate package
- instead of throwing an exception return a boolean true/false indicating if the script is valid

Once https://github.com/NationalSecurityAgency/datawave/pull/1274 is merged we can add this into the chain of validating visitors that are called

Reply:
Can do.